### DISCUSSION TITLE
GDS Parsing error: Could not find type MyEnumType

### DISCUSSION THREAD

**Discussion by JackNWhite** [nasa/fprime#1643] [Q&A] [answered]
I start fprime-gds, but get this response in the terminal:
```
[INFO] Ensuring TCP Server is stable for at least 5 seconds
[INFO] Running Application: /usr/bin/python3
[INFO] Log File: /home/jack/workspaces/tests/.........../ThreadedTCP.log
[INFO] Ensuring comm[ip] Application is stable for at least 1 seconds
[INFO] Running Application: /usr/bin/python3
[INFO] Ensuring HTML GUI is stable for at least 2 seconds
[INFO] Running Application: /usr/bin/python3
 * Serving Flask app "fprime_gds.flask.app"
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
[ERROR] Parsing error: Could not find type MyEnumType
[ERROR] Failed to run application: /usr/bin/python3 -u -m flask run --host 0.0.0.0 --port 5050. Error: /usr/bin/python3 stopped with code 1 sooner than 2 seconds.
[INFO] Shutting down F prime due to error. Failed to run HTML GUI
```

`MyEnumType` is specified in `MyEnumTypeEnumAi.xml`. I have manually validated it against **enum_schema.rng**, and my binary is built just fine. The enum appears in the dictionary in the build artifacts.

The enum is formatted thus:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="../Fprime/Autocoders/Python/schema/default/enum_schema.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>

<enum name="MyEnumType" default="RELINQUISH_BUFFER" namespace="Types">
    <comment>Location at which the event arose.</comment>
    
    <item name="RELINQUISH_BUFFER"                value="0"  />
    <item name="NOT_IMPLEMENTED"                  value="1"  />
....
</enum>
```

Have you had this problem? What was the solution?

**Comment by JackNWhite** [comment, marked as answer]
The answer was that, at one point of use in an XML file, the namespace was not specified.

**Comment by JackNWhite** [comment]
However, I do now have a second instance with a different (hand-coded) type in which the namespace *is* specified. I am aware that GDS doesn't do handcoded types, but shouldn't at least start up?

**Reply by Joshua-Anderson** [reply]
What were you using handcoded types for? Events, Telemetry, Commands, or all of the above? I agree that ideally the GDS should warn and ignore unsupported types rather than fail to start

**Reply by JackNWhite** [reply]
Event telemetry, for downlinking structures and arrays of a variable length set outside our F' application.

**Reply by Joshua-Anderson** [reply]
I opened a GDS issue to track this: https://github.com/fprime-community/fprime-gds/issues/96
